test/unit/pipeline/FetchUnitTest.cc

```diff
@@ -65,7 +68,7 @@ class PipelineFetchUnitTest : public testing::Test {
 
 // Tests that ticking a fetch unit attempts to predecode from the correct
 // program counter and generates output correctly.
-TEST_F(PipelineFetchUnitTest, Tick) {
+TEST_P(PipelineFetchUnitTest, Tick) {
   MacroOp macroOp = {uopPtr};
 
   ON_CALL(memory, getCompletedReads()).WillByDefault(Return(completedReads));
```

Comment:
I can't comment on the exact line, but the EXPECT_CALL below for predecode should return `insnMinSizeBytes` or `insnMaxSizeBytes`

Reply:
Additionally the output slot should be checked as to whether the expected mOp os present. It may be worth checking this with all the tests as it seems the original implementation was lacking in this regard